### PR TITLE
Move asset_tracker from 'samples/' folder to 'applications/'

### PR DESCRIPTION
**Pull request by jtguggedal**
Moving asset tracker to `applications/` to reflect that it's intended to be a complete application example, and not a sample showing usage of one certain functionality. 

I'm adding DNM in the title, as I'm not sure if CI is ready for this, and not sure what has to be changed in the docs.

-----
Move the asset_tracker from samples/ folder to applications/,
as it's intended to be a complete application example.

Signed-off-by: Jan Tore Guggedal <jantore.guggedal@nordicsemi.no>

### CODE REVIEW

**Comment by joakimtoe** [comment]
@ru-fu Does it require any doc change?

**Comment by jtguggedal** [comment]
@carlescufi @lemrey I'm not sure if the CI is ready for this move? We should probably hold off merging if it isn't.

**Comment by ru-fu** [comment]
Docs updated - please review (@carlescufi - please check the distinction between application and sample).
Output should be on \\hizz\components\ncs_docs\PR-457 in a bit.

File: doc/nrf/applications.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
+.. _applications:
+
+Applications
+############
+
+Applications consist of a fully integrated stack and interface drivers and libraries from the |NCS| and Zephyr.
```
Comment (carlescufi) [inline doc/nrf/applications.rst]: Applications typically include a fully integrated software stack, libraries and interface drivers from the |NCS| and its set of repositories.

File: doc/nrf/applications.rst [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+They implement a specific use case and can serve as a starting point for developing your product.
+
+Some applications come with a custom hardware.
```
Comment (carlescufi) [inline doc/nrf/applications.rst]: Some applications target custom hardware designed for a specific use-case, although they can also be run on different hardware like a generic development kit, albeit with limited functionality.

**Comment by carlescufi** [comment]
@ru-fu 	please take a look

File: doc/nrf/applications.rst (hunk 1 above)
Comment (ru-fu) [reply doc/nrf/applications.rst]: Not sure I understand ... Do you mean "they include a stack + libraries + interface drivers"?
I read the input as "they include a full stack, and they (resp. the stack) interface the following things: drivers and libraries".

Comment (carlescufi) [reply doc/nrf/applications.rst]: ah sorry, yes, I mean: "They include (i.e. in the build) stack + libraries + interface drivers"

**Comment by ru-fu** [comment]
OK, updated again @carlescufi

**Comment by myrdyr** [comment]
CI recursively looks for "sample.yaml" files in the "samples" directory, builds them, and uses hardcoded paths to all artifacts we decide that we want.

The Jenkinsfile should be updated to set testcase-root to $WORKSPAEC/nrf instead. This will probably expand the file paths from `zephyr/sanity-out/<board>/nrf9160/<samplename>/...` to `zephyr/sanity-out/<board>/<foldername>/nrf9160/<samplename>/...` where `foldername` is either 'applications' or 'samples' depending on what is being built.

To improve upon this, we could make the build process spit out something that's not called "zephyr.hex" or provide a better way to recognize what we want to keep as artifacts.

**Comment by jtguggedal** [comment]
@lemrey Rebased, but it seems the CI is red still. In Jenkins everything looks green to me, so not sure where to look. @ru-fu Could it be related to docs?

@carlescufi @myrdyr I'm not familiar enough with Jenkins to make this changes, I think. If we want it in this PR, please feel free to add a commit. I assume there are more changes coming in CI with regards to file structure changes in the near future, so perhaps it should go in a separate PR?

**Comment by ru-fu** [comment]
The doc build is green - so it shouldn't be related to the docs.

**Comment by carlescufi** [comment]
@jtguggedal I will fix CI, give me some time

**Comment by jtguggedal** [comment]
@carlescufi Great, thanks!

**Comment by lemrey** [comment]
@myrdyr can't see why CI is red on this one, could you take a look please?

**Comment by carlescufi** [comment]
@lemrey fixed it now

**Comment by myrdyr** [comment]
Carles already fixed it, but we are explicitly failing any build where we don't produce the required samples. This is why some builds are 'green', but end up as a failure. You can see this as a message during the sample build where it says "Sample xxx failed to build" or similar. Without this safe-guard, this PR would be green and and we'd also have stopped building asset_tracker in the future - without any warnings.

**Pull request merged**